### ISSUE TITLE
Show measurements query returns no results with an empty "default" RP

### ISSUE THREAD

**Issue by caitlin615** [influxdata/influxdb#6716]
### Bug report

__System info:__ v0.12 and v0.13

__Steps to reproduce:__

1. Start with a blank database
2. Create a new `DEFAULT` retention policy
  * `CREATE RETENTION POLICY rp_90d ON db DURATION 90d REPLICATION 1 DEFAULT`
3. No data or measurements in custom `DEFAULT` retention policy
4. Add a point to a new measurement (for example, `newmeasurement`) in the built-in `"default"` (now, not `DEFAULT`) retention policy.
5. Run `SHOW MEASUREMENTS` or `SHOW MEASUREMENTS WITH MEASUREMENT = newmeasurement`
6. No results returned
7. Once you add a point to the `DEFAULT` retention policy, the query returns all measurements as expected
 
__Expected behavior:__ Returns all measurements

__Actual behavior:__ Returns no results

__Additional info:__ 

Seems that `SHOW MEASUREMENTS` is rewriting this query to `SELECT _name AS "name" FROM db.rp_90d._measurements`, but it should really be rewriting it to `SELECT _name AS "name" FROM db."default"._measurements`.

The `SHOW MEASUREMENTS` query should either take a retention policy argument or be smart enough to check other retention policies

**Comment by shannonliu0809** [comment]
I am facing the same issue. 

I am able to insert data under one series name with different retention policies. #3341 I believe is also talking about the same issue.

I created a series called disk_free_2 with retention policy test.
`curl -X POST 'http://localhost:8086/write?db=shannon&rp=test' --data-binary 'disk_free_2,hostname=server01 value=123 1435362189575692182'`

then I created another series with the same name, disk_free_2, with a different retention policy, AggregatedRetention
`curl -X POST 'http://localhost:8086/write?db=shannon&rp=AggregatedRetention' --data-binary 'disk_free_2,hostname=server01 value=1 1435362189575692182'`

I only see ONE series when i do `show series`, but I am able to query the data 
`select * from AggregatedRetention.disk_free_2` and `select * from test.disk_free_2`

and they are indeed 2 different series.

show series should also include the rp associated with it.
